tests/api_test.py

```diff
@@ -69,3 +69,21 @@ def test_ignore_absolute_paths():
     assert get_path_if_relative(
         '/usr/lib/shared/schema.json',
     ) is None
+
+
+def test_resolve_nested_refs():
+    """
+    Make sure we resolve nested refs gracefully and not get lost in
+    the recursion. Also make sure we don't rely on dictionary order
+    """
+    import os
+    import yaml
+    from bravado_core.spec import Spec
+    from random import randint
+    from pyramid_swagger.api import resolve_refs
+    seed = randint(0, 4294967295)
```

Comment:
So the big issue was that the test fails for some orderings of dictionaries, the random was an attempt to make sure that with the patch, no ordering of the dictionary fails, but tbh having a random seed is exactly equivalent to not setting a seed at all :P. I'll remove it.